vertical-pod-autoscaler/initializer/initializer.go

```
@@ -0,0 +1,347 @@
+					pod.Name, container.Name, resource, recommended)
+			}
+
+			container.Resources.Requests[resource] = recommended
+		}
+	}
+
+}
+
+// applyPolicy updates recommendation if recommended resources exceed limits defined in VPA resources policy
+func applyPolicy(recommendation *apimock.ContainerRecommendation, policy *apimock.ContainerPolicy) {
+	for resourceName, recommended := range recommendation.Resources {
+		if policy != nil {
+			resourcePolicy, found := policy.ResourcePolicy[resourceName]
+			if found {
+				if !resourcePolicy.Min.IsZero() && recommended.Value() < resourcePolicy.Min.Value() {
+					glog.Warningf("recommendation outside of policy bounds : min value : %v recommended : %v",
+						resourcePolicy.Min.Value(), recommended)
+					recommendation.Resources[resourceName] = resourcePolicy.Min
+				}
+				if !resourcePolicy.Max.IsZero() && recommended.Value() > resourcePolicy.Max.Value() {
+					glog.Warningf("recommendation outside of policy bounds : max value : %v recommended : %v",
+						resourcePolicy.Max.Value(), recommended)
+					recommendation.Resources[resourceName] = resourcePolicy.Max
+				}
+			}
+		}
+
+	}
+}
+
+func getRecommendationForContainer(recommendation *apimock.Recommendation, container v1.Container) *apimock.ContainerRecommendation {
+	for i, containerRec := range recommendation.Containers {
+		if containerRec.Name == container.Name {
+			return &recommendation.Containers[i]
+		}
+	}
+	return nil
+}
+
+func getContainerPolicy(containerName string, policy *apimock.ResourcesPolicy) *apimock.ContainerPolicy {
+	if policy != nil {
+		for _, container := range policy.Containers {
+			if containerName == container.Name {
+				return &container
+			}
+		}
+	}
+	return nil
+}
+
+// This should be cached as part of vpaLister.
+func (i *initializer) getMatchingVPA(pod *v1.Pod) *apimock.VerticalPodAutoscaler {
+	configs, err := i.vpaLister.List()
+	if err != nil {
+		//cannot get vpa configs
```

probably not a good idea to eat all errors? wont list return other errors than not found?